core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -142,20 +142,16 @@ private[log] class ProducerIdEntry(val producerId: Long, val batchMetadata: muta
  *                                of this is the consumer offsets topic which uses producer ids from incoming
  *                                TxnOffsetCommit, but has no sequence number to validate and does not depend
  *                                on the deduplication which sequence numbers provide.
- * @param loadingFromLog This parameter indicates whether the new append is being loaded directly from the log.
- *                       This is used to repopulate producer state when the broker is initialized. The only
- *                       difference in behavior is that we do not validate the sequence number of the first append
- *                       since we may have lost previous sequence numbers when segments were removed due to log
- *                       retention enforcement.
+ * @param acceptFirstAppendUnconditionally
```

Comment:
We should document this parameter. Maybe some/all of the code comment could be moved here since it's more generally useful (imagine someone trying to decide if they should set this to true or false or reading code that sets this).